pkg/clustermanager/cluster_creator_test.go

```diff
@@ -0,0 +1,69 @@
+package clustermanager_test
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/pointer"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clustermanager"
+	"github.com/aws/eks-anywhere/pkg/clustermanager/mocks"
+	mockswriter "github.com/aws/eks-anywhere/pkg/filewriter/mocks"
+	mockskubeconfig "github.com/aws/eks-anywhere/pkg/kubeconfig/mocks"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+type clusterCreatorTest struct {
+	*WithT
+	ctx              context.Context
+	spec             *cluster.Spec
+	mgmtCluster      *types.Cluster
+	applier          *mocks.MockClusterApplier
+	writer           *mockswriter.MockFileWriter
+	kubeconfigWriter *mockskubeconfig.MockWriter
+}
+
+func newClusterCreator(t *testing.T) (*clustermanager.ClusterCreator, *clusterCreatorTest) {
+	ctrl := gomock.NewController(t)
+	cct := &clusterCreatorTest{
+		WithT:            NewWithT(t),
+		applier:          mocks.NewMockClusterApplier(ctrl),
+		writer:           mockswriter.NewMockFileWriter(ctrl),
+		kubeconfigWriter: mockskubeconfig.NewMockWriter(ctrl),
+		spec:             test.NewClusterSpec(),
+		ctx:              context.Background(),
+		mgmtCluster: &types.Cluster{
+			KubeconfigFile: "my-config",
+		},
+	}
+
+	cc := clustermanager.NewClusterCreator(cct.applier, cct.kubeconfigWriter, cct.writer)
+
+	return cc, cct
+}
+
+func (cct *clusterCreatorTest) expectFileCreate() {
+	cct.writer.EXPECT().Create(gomock.Any(), gomock.Any(), gomock.Any()).Return(os.NewFile(uintptr(*pointer.Uint(1)), "test"), "", nil)
+}
+
+func (cct *clusterCreatorTest) expectWriteKubeconfig() {
+	cct.kubeconfigWriter.EXPECT().WriteKubeconfig(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
+}
+
+func (cct *clusterCreatorTest) expectApplierRun() {
+	cct.applier.EXPECT().Run(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
```

Comment:
is there a way we can expect a bit more than any?
there is a bit more logic in this code than just calling those methods
The methods need to be called with the right arguments

Reply:
I'm not sure how to expect something specific for second and third arguments for filewriter's Create method, but I changed everything else.